hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/cdc/HoodieCDCRDD.scala

```diff
@@ -318,7 +318,7 @@ class HoodieCDCRDD(
               val after = record.get(3).asInstanceOf[GenericRecord]
               recordToLoad.update(3, convertToUTF8String(HoodieCDCUtils.recordToJson(after)))
             case HoodieCDCSupplementalLoggingMode.WITH_BEFORE =>
-              val row = cdcRecordDeserializer.deserialize(record).get.asInstanceOf[InternalRow]
+              val row = cdcRecordDeserialize(record)
```

Comment:
We actually don't need to copy it here right away: if someone just iterates over these records and simply writes them out into a file there would be no issue (and there's no need to make additional copies).

However if we have a place where we do retain reference after iteration, like:

```
val rows = iter.collect()
```

Then we will need to make copies before we do that, like below:

```
val copiedRows = iter.map(_.copy()).collect()
```